arctic/chunkstore/chunkstore.py

```diff
@@ -266,67 +308,58 @@ def write(self, symbol, item, chunk_size):
-        if sym[TYPE] == 'dataframe' and not isinstance(item, DataFrame):
-            raise Exception("Cannot combine DataFrame and Series")
+        symbol = sym[SYMBOL]
 
         if chunk_range is not None:
             self.delete(symbol, chunk_range)
             sym = self._get_symbol_info(symbol)
 
         bulk = self._collection.initialize_unordered_bulk_op()
         op = False
-        for start, end, record in self.chunker.to_chunks(item, sym[CHUNK_SIZE]):
-            # read out matching chunks
-            df = self.read(symbol, chunk_range=self.chunker.to_range(start, end), filter_data=False)
+        chunker = CHUNKER_MAP[sym[CHUNKER]]
 
+        for start, end, _, record in chunker.to_chunks(item, chunk_size=sym[CHUNK_SIZE]):
+            # read out matching chunks
+            df = self.read(symbol, chunk_range=chunker.to_range(start, end), filter_data=False)
             # assuming they exist, update them and store the original chunk
             # range for later use
-            if not df.empty:
-                record = combine_method(record, df)
+            if len(df) > 0:
+                record = combine_method(df, record)
                 if record is None or record.equals(df):
                     continue
 
                 sym[APPEND_COUNT] += len(record)
-                sym[ROWS] += len(record) - len(df)
-                new_chunk = False
+                sym[LEN] += len(record) - len(df)
             else:
-                new_chunk = True
                 sym[CHUNK_COUNT] += 1
-                sym[ROWS] += len(record)
+                sym[LEN] += len(record)
 
-            data = self.serializer.serialize(record)
+            data = SER_MAP[sym[SERIALIZER]].serialize(record)
             op = True
 
-            chunk = {DATA: data}
-            chunk[TYPE] = 'dataframe' if isinstance(record, DataFrame) else 'series'
-            chunk[START] = start
-            chunk[END] = end
-            sha = self._checksum(chunk)
-            chunk[SHA] = sha
-            if new_chunk:
-                # new chunk
-                bulk.find({SYMBOL: symbol, SHA: sha}
+            size_chunked = len(data[DATA]) > MAX_CHUNK_SIZE
+            for i in xrange(int(len(data[DATA]) / MAX_CHUNK_SIZE + 1)):
+                chunk = {DATA: Binary(data[DATA][i * MAX_CHUNK_SIZE : (i + 1) * MAX_CHUNK_SIZE])}
+                chunk[METADATA] = data[METADATA]
+                if size_chunked:
+                    chunk[SEGMENT] = i
+                else:
+                    chunk[SEGMENT] = -1
+                chunk[START] = start
+                chunk[END] = end
+                chunk[SYMBOL] = symbol
+                dates = [chunker.chunk_to_str(start), chunker.chunk_to_str(end), str(chunk[SEGMENT]).encode('ascii')]
+                sha = self._checksum(dates, data[DATA])
+                chunk[SHA] = sha
+                bulk.find({SYMBOL: symbol, START: start, END: end, SEGMENT: chunk[SEGMENT]}
```

Comment:
What happens if someone does a chunk update, the old chunk is segmented, and the new chunk has fewer segments than the old one?  I think we'll have a rogue segment remaining?